### PR TITLE
Version 0 0 2

### PR DESCRIPTION
**Pull request by rmeissner**

### CODE REVIEW

File: source/contracts/architecture.md [hunk 1]
```diff
@@ -67,20 +67,22 @@ Before a Safe transaction can be executed, the transaction has to be confirmed b
...
+There are different types of signatures:
 
-#### GnosisSafeTeamEdition.sol
-This version is targeted at teams where each owner is a different user. Each owner has to confirm a transaction by using `approveTransactionWithParameters`. Once the required number of owners has confirmed, the transaction can be executed via `execTransactionIfApproved`. If the sender of `execTransactionIfApproved` is an owner it is not necessary to confirm the transaction before. Furthermore this version doesn't store the nonce in the contract but for each transaction a nonce needs to be specified.
+1. ECDSA signatures generated by Externally Owned Accounts
+1. EIP-1271 based contract signatures
+1. Preferified transaction hash signatures
 
-For more information check out the section about the [Gnosis Safe Team Edition](./team_edition.html).
+For more information check on how the signature types and how to generate and encode the see [Signatures](./signatures.html).
```
Comment (tschubotz) [inline source/contracts/architecture.md]: This sentence sounds weird. Should this be something like `For more information on  signature types including how to generate and encode them, see [Signatures](./signatures.html).`?

File: source/contracts/architecture.md [hunk 2]
```diff
@@ -67,20 +67,22 @@ Before a Safe transaction can be executed, the transaction has to be confirmed b
...
+To execute a transaction the method `execTransaction` can be used. To approve a transaction it is necessary to generate and encode the required signatures. 
 
-For more information check out the section about the [Gnosis Safe Personal Edition](./personal_edition.html).
+There are different types of signatures:
 
-#### GnosisSafeTeamEdition.sol
-This version is targeted at teams where each owner is a different user. Each owner has to confirm a transaction by using `approveTransactionWithParameters`. Once the required number of owners has confirmed, the transaction can be executed via `execTransactionIfApproved`. If the sender of `execTransactionIfApproved` is an owner it is not necessary to confirm the transaction before. Furthermore this version doesn't store the nonce in the contract but for each transaction a nonce needs to be specified.
+1. ECDSA signatures generated by Externally Owned Accounts
+1. EIP-1271 based contract signatures
+1. Preferified transaction hash signatures
```
Comment (tschubotz) [inline source/contracts/architecture.md]: Preferred?

File: source/contracts/deployment.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
+# Deployment
+The Gnosis Safe Personal Edition smart contract was written with the usage of a proxy contract in mind. Because of that there is no constructor and it is required to call an inilize function on the contract before it can be used. For this it is recommended to use the ProxyFactory or the DelegatingConstructorProxy.
```
Comment (tschubotz) [inline source/contracts/deployment.md]: ~Personal edition~ ?

Comment (tschubotz) [reply source/contracts/deployment.md]: `s/inilize/initialize`

File: source/contracts/architecture.md (hunk 2 above)
Comment (rmeissner) [reply source/contracts/architecture.md]: Pre-verified :P

File: source/contracts/transactions.md [hunk 4]
```diff
@@ -0,0 +1,102 @@
+# Transaction Execution
+To execute a transaction with the Gnosis Safe Personal Editon the `execTransaction` methods needs to be called with the following parameters:
```
Comment (tschubotz) [inline source/contracts/transactions.md]: Did you mean to mention "Personal Editon" specifically?

File: source/contracts/meta_tx.md [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+As all the parameters required for execution are part of the submitted transaction it is possible that miners front-run the original relayer to receive the reward. In the long run that behaviour would be appreciated, since it would allow that anybody submits these transactions with `gasPrice` of the transaction triggering `execTransaction` set to 0. Miners could pick up these transactions and claim the rewards. This is why it is possible to specify the account that should receive the refund.
+
+The vision for submitting Safe transactions would be the following:
+
+1. Acquire estimates for a Safe transaction from potential relays
+1. Choose a relay and generate the signatures for the Safe transaction (the choosen relay should be set to receive the rewards to prevent front-running)
+1. Submit the Safe transaction with all required paramters to the choosen relayer.
+
+Currently this implemented a simple REST API. But the idea would be to make use of an open decentralized system that supports different relayers (see [MetaCartel](https://github.com/Meta-tx)).
```
[the remark this answers is not in the document]
Comment (rmeissner) [reply source/contracts/meta_tx.md]: as a ;)

**Pull request merged**